reorder_python_imports/main.py

```diff
@@ -247,12 +254,30 @@ def apply_import_sorting(partitions, separate_relative=False, **sort_kwargs):
     new_imports = []
     relative_imports = []
 
+    def _is_require_extra_new_line(last_import_obj, import_obj):
```

Comment:
This looks awesome, but why would you call this `_import_type_switches`? I think that's is not clear enough to describe what this function does.